ubuntu-mainline-kernel.sh

```diff
@@ -369,6 +369,22 @@ latest_local_version() {
 }
 
 remote_html_cache=""
+version_parse() {
+    local line
+    while read -r line; do
+        if [[ $line =~ DIR.*href=\"(v[[:digit:]]+\.[[:digit:]]+(\.[[:digit:]]+)?)(-(rc[[:digit:]]+))?/\" ]]; then
+            line="${BASH_REMATCH[1]}"
+            if [[ -z "${BASH_REMATCH[2]}" ]]; then
+                line="$line.0"
+            fi
+            if [[ -n "${BASH_REMATCH[3]}" ]]; then
```

Comment:
Please add a code comment about replacing the `-` with `~` for rc versions, eg Im not fully sure why this is needed?

Reply:
GNU version sort treats `~` specially for suffixes, placing them before versions that are otherwise the same. This is how the problem of `vX.0` coming before `vX.0-rc1` is resolved. I can comment this better.